pybossa/api/__init__.py

```diff
@@ -355,6 +359,63 @@ def user_progress(project_id=None, short_name=None):
     else:  # pragma: no cover
         return abort(404)
 
+@jsonpify
+@blueprint.route('/app/<short_name>/taskprogress')
+@blueprint.route('/project/<short_name>/taskprogress')
+@blueprint.route('/app/<int:project_id>/taskprogress')
+@blueprint.route('/project/<int:project_id>/taskprogress')
+@ratelimit(limit=ratelimits.get('LIMIT'), per=ratelimits.get('PER'))
+def task_progress(project_id=None, short_name=None):
+    """API endpoint for task progress.
+
+    Returns a JSON object continaing the number of tasks which meet the user defined filter constraints
+    """
+    if current_user.is_anonymous:
+        return abort(401)
+    if not (project_id or short_name):
+        return abort(404)
+    if short_name:
+        project = project_repo.get_by_shortname(short_name)
+    elif project_id:
+        project = project_repo.get(project_id)
+
+    filter_fields = request.args
+
+    if not project:
+        return abort(404)
+
+    sql_text = "SELECT COUNT(*) FROM task WHERE project_id=" + str(project.id)
+
+    task_info_fields = get_searchable_columns(project.id)
+    task_fields = [
+        "id",
+        "state",
+        "n_answers",
+        "created",
+        "calibration",
+    ]
```

Comment:
this can be outside of function

Reply:
should this go on the top of the file below headers? I'm not familiar when it's best to separate the list of fields in this case from the method.